plugins/net_plugin/net_plugin.cpp

```diff
@@ -598,18 +598,25 @@ namespace eosio {
    // thread safe
    class queued_buffer : boost::noncopyable {
    public:
+      void reset() {
+         fc::lock_guard g( _mtx );
+         _write_queue.clear();
+         _trx_write_queue.clear();
+         _write_queue_size = 0;
+         _out_queue.clear();
+      }
+
       void clear_write_queue() {
          fc::lock_guard g( _mtx );
          _write_queue.clear();
-         _sync_write_queue.clear();
+         _trx_write_queue.clear();
          _write_queue_size = 0;
       }
```

Comment:
`clear_write_queue()` and `reset()` have most of code in common.

Reply:
The mutex means we would need an internal private method. The code duplication here seemed to me to be clearer than that.